src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindIntegrationTest.java

```diff
@@ -772,5 +780,139 @@ public void findByBooleanColumn() {
+          .body("data.docs", jsonEquals(arrayNode.toString()));
+    }
+
+    @Test
+    @Order(4)
+    public void sortWithSkipLimit() {
+      deleteAllDocuments();
+      Map<String, String> sorted = getDocuments(25, true);
+      insert(sorted);
+      String json =
+          """
+          {
+            "find": {
+              "sort" : ["username"],
+              "options" : {"skip": 10, "limit" : 10}
+            }
+          }
+          """;
+
+      JsonNodeFactory nodefactory = objectMapper.getNodeFactory();
+      final ArrayNode arrayNode = nodefactory.arrayNode(10);
+      final Iterator<Map.Entry<String, String>> iterator = sorted.entrySet().iterator();
+      try {
+        for (int i = 0; i < 20; i++) {
+          String value = iterator.next().getValue();
+          if (i >= 10) {
+            arrayNode.add(objectMapper.readTree(value));
+          }
+        }
+      } catch (Exception e) {
+        // ignore the object node creation error should never happen
+      }
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.count", is(10))
+          .body("data.docs", jsonEquals(arrayNode.toString()));
+    }
+
+    @Test
+    @Order(5)
+    public void sortDescending() {
+      deleteAllDocuments();
+      Map<String, String> sorted = getDocuments(25, false);
+      insert(sorted);
+      String json =
+          """
+          {
+            "find": {
+              "sort" : ["-username"]
+            }
+          }
+          """;
+
+      JsonNodeFactory nodefactory = objectMapper.getNodeFactory();
+      final ArrayNode arrayNode = nodefactory.arrayNode(20);
+      final Iterator<Map.Entry<String, String>> iterator = sorted.entrySet().iterator();
+      try {
+        for (int i = 0; i < 20; i++)
+          arrayNode.add(objectMapper.readTree(iterator.next().getValue()));
+      } catch (Exception e) {
+        // ignore the object node creation error should never happen
+      }
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, keyspaceId.asInternal(), collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.count", is(20))
+          .body("data.docs", jsonEquals(arrayNode.toString()));
+    }
+
+    private void insert(Map<String, String> documents) {
+      documents.values().forEach(doc -> insertDoc(doc));
+    }
+
+    private Map<String, String> getDocuments(int countOfDocuments, boolean asc) {
+      String json = "{\"_id\":\"doc%s\", \"username\":\"user%s\", \"active_user\":true}";
+      Map<String, String> data =
```

Comment:
Is the idea here to keep the order same, or opposite? If we wanted to avoid something that is not sorted (to verify sorting), could just use basic `HashMap`. It should still produce repeatable runs, I think; order would be arbitrary but stable (based on hashCode() of `String`).

Reply:
Wanted to keep the data sorted in this response. This ordered value is used to check against the find command response to confirm sort.